### PR TITLE
kubeadm: add mandatory phase "annotate-cri"

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
In v1.12, after the changing how kubeadm-config upload/fetch, AnnotateCRISocket is now a necessary step for getting a cluster that can be upgraded using phases. This PR adds a new subcommand for letting users to invoke AnnotateCRISocket as a phase

**Which issue(s) this PR fixes**:
Fixes # https://github.com/kubernetes/kubeadm/issues/925

**Special notes for your reviewer**:
When testing this PR, I found that one of the tests on component config defaulting was broken (didn't have time to investigate why yet). This PR fixes this test as well  

**Release note**:
```release-note
kubeadm: added phase command "alpha phase kubelet config annotate-cri"
```
@kubernetes/sig-cluster-lifecycle-pr-reviews
/sig cluster-lifecycle
/area kubeadm
/assign @timothysc @neolit123 
/kind bug

### CODE REVIEW

File: cmd/kubeadm/app/util/config/testdata/defaulting/master/defaulted.yaml [hunk 1]
```diff
@@ -115,7 +115,6 @@ evictionHard:
   imagefs.available: 15%
   memory.available: 100Mi
   nodefs.available: 10%
-  nodefs.inodesFree: 5%
```
Comment (neolit123) [inline cmd/kubeadm/app/util/config/testdata/defaulting/master/defaulted.yaml]: possibly a stray change?

**Review by neolit123 — APPROVED** [review]
thanks @fabriziopandini LGTM

**Review by timothysc — APPROVED** [review]
/approve 

I'll hold for other feedback on UX. 
/cc @kubernetes/sig-cluster-lifecycle-pr-reviews

File: cmd/kubeadm/app/cmd/phases/kubelet.go [hunk 2]
```diff
@@ -222,6 +232,45 @@ func NewCmdKubeletConfigUpload() *cobra.Command {
...
+		Long:    kubeletConfigAnnotateCRILongDesc,
+		Example: kubeletConfigAnnotateCRIExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			if len(cfgPath) == 0 {
+				kubeadmutil.CheckErr(fmt.Errorf("The --config argument is required"))
+			}
+
+			// KubernetesVersion is not used, but we set it explicitly to avoid the lookup
+			// of the version from the internet when executing ConfigFileAndDefaultsToInternalConfig
+			err := SetKubernetesVersion(nil, cfg)
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/phases/kubelet.go]: sorry, i was so tired last night that i might have missed this.

i'm pretty sure that ConfigFileAndDefaultsToInternalConfig wil end up calling this:
https://github.com/kubernetes/kubernetes/blob/ed958b7d79ee7eee0d71952e71475fe667c77909/cmd/kubeadm/app/util/version.go#L62

which handles the look up from the internet. it tries it, but then it falls back to the client version in air-gapped cases.

SetKubernetesVersion() on the hand falls back to the client version directly:
https://github.com/kubernetes/kubernetes/blob/7f23a743e8c23ac6489340bbb34fa6f1d392db9d/cmd/kubeadm/app/phases/upgrade/versiongetter.go#L74

i guess it's a question if want to always first fall back to the client version for this phase?

Comment (fabriziopandini) [reply cmd/kubeadm/app/cmd/phases/kubelet.go]: IMO Let's gets this into beta-2 an then eventually iterate for the air gapped/offline cases

Comment (neolit123) [reply cmd/kubeadm/app/cmd/phases/kubelet.go]: ok, sounds good.
i guess my point was that the call to `SetKubernetesVersion` might be redundant, but i haven't verified.

**Pull request merged**